lib/system.ml

```diff
@@ -302,7 +302,7 @@ let with_time ~batch f x =
     raise e
 
 let get_toplevel_path top =
-  let dir = Filename.dirname Sys.argv.(0) in
+  let dir = Filename.dirname Sys.executable_name in
```

Comment:
Uh, have you tested this with bytecode builds of all the utilities, including coqide + workers?

Reply:
>Uh, have you tested this with bytecode builds of all the utilities, including coqide + workers?

I don't know how to do that.

Reply:
Build a byte-only version of Coq and use `coqide.byte` , but even with the current config `coqide.byte` this to call `coqidetop.byte`, but that's always a debate given that client/servers are compatible in that sense.

Reply:
>even with the current config coqide.byte this to call coqidetop.byte

What? This sentence is ill-formed.

Reply:
`coqide.byte` tries to call `coqidetop.byte`, so I mean you usually don't need to ensure you are in a byte-only build.

Reply:
But `Sys.executable_name` I think tends to be `ocaml` for bytecode binaries, so beware of that.

Reply:
```
gaetan@debian-hibiki ~/d/c/coq> bin/coqide.byte &
gaetan@debian-hibiki ~/d/c/coq> ps aux | grep coq
gaetan   17054  3.4  0.5 448760 60196 pts/4    Sl   15:29   0:00 /home/gaetan/.opam/4.06.0/bin/ocamlrun bin/coqide.byte
gaetan   17061  1.0  0.4 712772 55672 pts/4    Sl   15:29   0:00 /home/gaetan/.opam/4.06.0/bin/ocamlrun bin/coqidetop.byte -main-channel stdfds -control-channel 127.0.0.1:36095:39245 --xml_format=Ppcmds -async-proofs on
gaetan   17065  1.1  0.4 622572 55376 pts/4    Sl   15:29   0:00 /home/gaetan/.opam/4.06.0/bin/ocamlrun bin/coqproofworker.byte -main-channel stdfds -control-channel 127.0.0.1:54771:47601 --xml_format=Ppcmds -worker-id proofworker:0 -async-proofs-worker-priority low
gaetan   17094  0.0  0.0  13948  1028 pts/4    S+   15:29   0:00 grep --color=auto coq

```

Reply:
Umm, if you are getting the directory where `ocamlrun` is, how can this work? Umm... I need to have a look.

Reply:
If you `let () = Printf.printf "%s\n" Sys.executable_name` it prints
```
bin/coqidetop.byte
Welcome to Coq 8.8+alpha (May 2018)
bin/coqproofworker.byte
bin/coqide.byte
```
ocamlrun is just ps being well informed.

Reply:
Umm, that used not to be the case in the past, let me check in some switches.